warehouse/router/upload.go

```diff
@@ -363,6 +364,9 @@ func (job *UploadJob) run() (err error) {
 			if err = job.exportData(); err != nil {
 				break
 			}
+			if err = job.cleanupObjectStorageFiles(); err != nil {
+				break
+			}
```

Comment:
@shekhar-rudder Say if this fails. Would the job be retried? If yes, in next retry would it continue from just `cleanupObjectStorageFiles`?

Reply:
@shekhar-rudder @achettyiitr Also from [PRD doc](https://www.notion.so/rudderstacks/Staging-Load-files-deletion-174f2b415dd080aba52bdecacad07191), did we close on failing the job or considering metric and internal alert for now?

Reply:
@RanjeetMishra If the cleanup fails, the job will be marked as failed. On the next retry, I would assume it will proceed as it would for any other error.

Reply:
Since now it is config based deletion, I think its fine to mark the job as failed. If customer doesn't want the syncs to error out, they can always turn it off.

Reply:
Seems fine if we are aligning with marking sync jobs as failed. Regarding destination config for old and new destinations lets finalize tomorrow. This PR is good to go in that case.